hashes/src/hmac/mod.rs

```diff
@@ -132,8 +132,10 @@ impl<T: GeneralHash> Hash for Hmac<T> {
 
     fn from_byte_array(bytes: T::Bytes) -> Self { Hmac(T::from_byte_array(bytes)) }
 
-    #[allow(deprecated)]
-    fn from_slice(sl: &[u8]) -> Result<Hmac<T>, FromSliceError> { T::from_slice(sl).map(Hmac) }
+    #[allow(deprecated_in_future)] // Because of `FromSliceError`.
```

Comment:
> The `deprecated_in_future` lint is internal to rustc and should not be used by user code.

https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.DEPRECATED_IN_FUTURE.html

Reply:
Lol wtf. "Don't use this" with no replacement or workaround or anything.Let's just keep using it.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/issues/4082

Reply:
It can be removed before release when we remove `TBD` - assuming we don't forget it when doing so.